src/backend/opengl/CommandBufferGL.cpp

```diff
@@ -85,9 +109,87 @@ namespace opengl {
                                 break;
                         }
                     }
+
+                    dirtyBits[stage].reset();
                 }
             }
+        };
+
+        // Vertex buffers and index buffers are implemented as part of an OpenGL VAO that corresponds to an
+        // InputState. On the contrary in NXT they are part of the global state. This means that we have to
+        // re-apply these buffers on an InputState change.
+        struct InputBufferTracker {
+            bool indexBufferDirty = false;
+            Buffer* indexBuffer = nullptr;
+
+            std::bitset<kMaxVertexAttributes> dirtyVertexBuffers;
```

Comment:
Why `kMaxVertexAttributes` here and `kMaxVertexInputs` below? Seems like they're both indexed the same way in `OnSetVertexBuffers`.

Reply:
Whoops, code was wrong. Done.